libr/anal/xrefs.c

```diff
@@ -251,15 +251,24 @@ R_API void r_anal_xrefs_list(RAnal *anal, int rad) {
 				} else {
 					anal->cb_printf (",");
 				}
-				anal->cb_printf ("\"%"PFMT64d"\":%"PFMT64d, ref->at, ref->addr);
+
+				char *name = anal->coreb.getNameDelta (anal->coreb.core, ref->at);
+				r_str_replace_char (name, ' ', 0);
+				anal->cb_printf ("{\"name\":\"%s\",", name? name: "");
+				free (name);
+				anal->cb_printf ("\"at\":%"PFMT64d",\"type\":\"%s\",\"address\":%"PFMT64d, ref->at, r_anal_xrefs_type_tostring (t), ref->addr);
+				name = anal->coreb.getNameDelta (anal->coreb.core, ref->addr);
+				r_str_replace_char (name, ' ', 0);
+				anal->cb_printf (",\"xref\":\"%s\"}", (name && *name) ? name : "");
```

Comment:
You are checking for name&&*name in 2 places in different ways. Maybe better to use r_str_get to unify behaviours